packages/core/src/http-client/http-client-types.ts

```diff
@@ -9,6 +9,25 @@ export interface DestinationHttpRequestConfig {
   httpsAgent?: http.Agent;
 }
 
+type Method =
+  | 'get'
+  | 'GET'
+  | 'delete'
+  | 'DELETE'
+  | 'head'
+  | 'HEAD'
+  | 'options'
+  | 'OPTIONS'
+  | 'post'
+  | 'POST'
+  | 'put'
+  | 'PUT'
+  | 'patch'
+  | 'PATCH';
+
+/**
+ * @deprecated Since v1.20.0. Use method string directly, e. g. 'get' or 'GET'.
```

Comment:
Why is this? Isn't it better to use a type for the methods see above?

Reply:
We say we provide an API that is compatible with axios. The type above is (almost) the same as in axios. The enum below is compatible but you need to use the enum and you need to import the httpmethod type. I decided that making the API more 'free' in that sense is better, that relying on an additional import, given that this is not a breaking change.

Reply:
Personally I do not see a big difference between a enum and a union type of strings. I agree that for the enum you need the import but therefore if you have a pure literal enum each enum becomes a type as well which can be useful.

Note that enums and types are relates - perhaps you can transform the enum in a type somehow without defining it:
```
enum LogLevel {
    ERROR, WARN, INFO, DEBUG
}

/**
 * This is equivalent to:
 * type LogLevelStrings = 'ERROR' | 'WARN' | 'INFO' | 'DEBUG';
 */
type LogLevelStrings = keyof typeof LogLevel;
```

Reply:
But what is the benefit? From a usage perspective it cannot get simpler than passing strings. We should definitely allow lowercase. For this I could extend the enum, but why would that be better?

Reply:
Based on your comment below, I will resolve this one.

Reply:
I meant perhaps you can use the existing enum to define the type. Less maintanace.